File: Java-SpringBoot/BE-02-DB/java-17-springboot-3.5/src/main/java/com/flyrank/taskservice/dto/StatsResponse.java
package com.flyrank.taskservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import java.util.List;

/**
 * Payload returned by GET /stats endpoint.
 */
@Schema(description = "Database statistics payload")
public record StatsResponse(
        @Schema(description = "Database table names", example = "[\"TASKS\"]")
        List<String> tables,

        @Schema(description = "Total number of tasks", example = "5")
        @JsonProperty("total_tasks")
        long totalTasks,

        @Schema(description = "Count of completed tasks", example = "2")
        @JsonProperty("done_tasks")
        long doneTasks,

        @Schema(description = "Count of open tasks", example = "3")
        @JsonProperty("open_tasks")
        long openTasks
) {}

File: Java-SpringBoot/BE-02-DB/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/taskservice/dto/StatsResponse.kt
package com.flyrank.taskservice.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Database statistics payload")
data class StatsResponse(
    @field:Schema(description = "Database table names", example = "[\"TASKS\"]")
    val tables: List<String>,

    @field:Schema(description = "Total number of tasks", example = "5")
    @field:JsonProperty("total_tasks")
    val totalTasks: Long,

    @field:Schema(description = "Count of completed tasks", example = "2")
    @field:JsonProperty("done_tasks")
    val doneTasks: Long,

    @field:Schema(description = "Count of open tasks", example = "3")
    @field:JsonProperty("open_tasks")
    val openTasks: Long
)
